pandas/io/excel/_base.py

```diff
@@ -658,6 +658,12 @@ class ExcelWriter(metaclass=abc.ABCMeta):
     datetime_format : str, default None
         Format string for datetime objects written into Excel files.
         (e.g. 'YYYY-MM-DD HH:MM:SS').
+    num_formats : dict, default None
```

Comment:
do we use this name in any other place? e call this `float_format` in the csv writer. i suppose `number_fromat` could be ok (prefer float_format if this is *only* for floats)

Reply:
This change is being done **only** for the `date`/`datetime` types for the time being, so I think I can call this `formatters` instead, as has been suggested by you in another comment.

Reply:
hidden in the user guide for Styler is the use of the pseudo CSS value `number-format`, for use with `Styler.to_excel()` to coerce certain cells to a specific Excel format. Albeit this is generally applied for any kind of dtype, and not specifically for `date` and `datetime`.

It seems confusing to have a `num_formats` arg which operates exclusively for dates and datetimes, and overwrites the args: `date_format` and `datetime_format`, wouldn't it be better to reorganise the existing args?

Reply:
Reorganizing the existing args is good but I preferred not to modify them but chose to go in for a new parameter to make it more generic if the parameter later on needs to be extended to other types like `int` and `float` as well.